### ISSUE TITLE
Rename variable should not trigger on change to discard

### ISSUE THREAD

**Issue by 333fred** [dotnet/roslyn#34280]
```C#
class C
{
    int M()
    {
        int i;
        i/*Change i here to _*/ = M();
        return i;
    }
}
```

When I change the `i` variable to `_` where indicated by the comment, rename offers to change the name of the variable to `_`. It should be intelligent around the discard identifier and not offer to rename in this scenario.
